### ISSUE TITLE
Segfault in raft code

### ISSUE THREAD

**Issue by apollo13** [hashicorp/nomad#7860]
### Nomad version
Nomad v0.11.0 (5f8fe0afc894d254e4d3baaeaee1c7a70a44fdc6)

### Operating system and Environment details
Debian 10

### Issue

Segfault during start of a single node:
```
May 04 18:17:18 nomad01 nomad[13433]: WARNING: keyring exists but -encrypt given, using keyring
May 04 18:17:18 nomad01 nomad[13433]: ==> WARNING: Bootstrap mode enabled! Potentially unsafe operation.
May 04 18:17:18 nomad01 nomad[13433]: ==> Loaded configuration from /opt/paas/etc/nomad/nomad.d/40_general.hcl, /opt/paas/etc/nomad/nomad.d/50_client.hcl, /opt/paas/etc/nomad/nomad.d/50_server.hcl
May 04 18:17:18 nomad01 nomad[13433]: ==> Starting Nomad agent...
May 04 18:17:20 nomad01 consul[523]:     2020-05-04T18:17:20.542Z [INFO]  agent: Synced check: check=_nomad-check-3fd6bef8a509de45c066ae463c50ceba82630433
May 04 18:17:20 nomad01 nomad[13433]: panic: runtime error: invalid memory address or nil pointer dereference
May 04 18:17:20 nomad01 nomad[13433]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x17f190b]
May 04 18:17:20 nomad01 nomad[13433]: goroutine 15 [running]:
May 04 18:17:20 nomad01 nomad[13433]: github.com/hashicorp/nomad/nomad/state.(*StateStore).deleteJobFromPlugin(0xc00072b710, 0x5457, 0xc000aa2300, 0xc000f78d20, 0xc000b9bec0, 0x0)
May 04 18:17:20 nomad01 nomad[13433]:         github.com/hashicorp/nomad/nomad/state/state_store.go:1192 +0x37b
May 04 18:17:20 nomad01 nomad[13433]: github.com/hashicorp/nomad/nomad/state.(*StateStore).DeleteJobTxn(0xc00072b710, 0x5457, 0xc000abaa96, 0x7, 0xc000abaa66, 0x7, 0xc000aa2300, 0x38061e0, 0xc000b9b760)
May 04 18:17:20 nomad01 nomad[13433]:         github.com/hashicorp/nomad/nomad/state/state_store.go:1502 +0xafb
May 04 18:17:20 nomad01 nomad[13433]: github.com/hashicorp/nomad/nomad.(*nomadFSM).handleJobDeregister(0xc0003409a0, 0x5457, 0xc000abaa66, 0x7, 0xc000abaa96, 0x7, 0x1, 0xc000aa2300, 0xc000a55a38, 0x17dbfc1)
May 04 18:17:20 nomad01 nomad[13433]:         github.com/hashicorp/nomad/nomad/fsm.go:618 +0x1f6
May 04 18:17:20 nomad01 nomad[13433]: github.com/hashicorp/nomad/nomad.(*nomadFSM).applyDeregisterJob.func1(0xc000aa2300, 0x3142c01, 0xc000aa2300)
May 04 18:17:20 nomad01 nomad[13433]:         github.com/hashicorp/nomad/nomad/fsm.go:565 +0x78
May 04 18:17:20 nomad01 nomad[13433]: github.com/hashicorp/nomad/nomad/state.(*StateStore).WithWriteTransaction(0xc00072b710, 0xc000a55b10, 0x0, 0x0)
May 04 18:17:20 nomad01 nomad[13433]:         github.com/hashicorp/nomad/nomad/state/state_store.go:4873 +0x7b
May 04 18:17:20 nomad01 nomad[13433]: github.com/hashicorp/nomad/nomad.(*nomadFSM).applyDeregisterJob(0xc0003409a0, 0xc0000b4301, 0x72, 0x72, 0x5457, 0x0, 0x0)
May 04 18:17:20 nomad01 nomad[13433]:         github.com/hashicorp/nomad/nomad/fsm.go:564 +0x1f2
May 04 18:17:20 nomad01 nomad[13433]: github.com/hashicorp/nomad/nomad.(*nomadFSM).Apply(0xc0003409a0, 0xc000536c30, 0x577acc0, 0xbfa434ec263f1609)
May 04 18:17:20 nomad01 nomad[13433]:         github.com/hashicorp/nomad/nomad/fsm.go:208 +0x42d
May 04 18:17:20 nomad01 nomad[13433]: github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft.(*Raft).runFSM.func1(0xc000ad1350)
May 04 18:17:20 nomad01 nomad[13433]:         github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft/fsm.go:90 +0x2c1
May 04 18:17:20 nomad01 nomad[13433]: github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft.(*Raft).runFSM.func2(0xc000dd7200, 0x3b, 0x40)
May 04 18:17:20 nomad01 nomad[13433]:         github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft/fsm.go:113 +0x75
May 04 18:17:20 nomad01 nomad[13433]: github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft.(*Raft).runFSM(0xc00054a300)
May 04 18:17:20 nomad01 nomad[13433]:         github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft/fsm.go:219 +0x42f
May 04 18:17:20 nomad01 nomad[13433]: github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc00054a300, 0xc0002cc8e0)
May 04 18:17:20 nomad01 nomad[13433]:         github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft/state.go:146 +0x55
May 04 18:17:20 nomad01 nomad[13433]: created by github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft.(*raftState).goFunc
May 04 18:17:20 nomad01 nomad[13433]:         github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft/state.go:144 +0x66
```

### Reproduction steps
No idea

**Comment by tgross** [comment]
Hi @apollo13! Sorry about this. It was fixed by https://github.com/hashicorp/nomad/pull/7758 and released in 0.11.1. If you upgrade to that, this should be fixed for you.

**Comment by apollo13** [comment]
Thank you, sorry I tried searching the issues but apparently didn't know what to search for.
